D.gitignore

```diff
@@ -0,0 +1,36 @@
+# Gitignore for projects in D using dub.
+# Dlang: http://dlang.org/
+# Dub:   https://github.com/D-Programming-Language/dub
+
+## Binary files ##
+
+# Object files
+*.o
+*.obj
+
+# Libraries
+*.a
+*.lib
+*.so
+*.dylib
+*.dll
+
+# Executables
+*.exe
+
+
+## Dub specific ##
+.dub/
+
+# Result of 'dub test'
+__test__*__
+
+# Documentation artifacts (dub --build=[docs|ddox])
+docs.json
+__dummy.html
+docs/*
+
+# By default, ignore dub.selections.json.
+# You should usually rely on tags in dub.json rather than pinning your project
+# to a certain commit, but if you really need to, remove this line:
+dub.selections.json
```

Comment:
You're clearly not aware of the reasons why `dub.selections.json` was added and why it's needed.

The problem is a scenario like this:

In you're project you have specified a dependency on project A with an exact version, "1.2.3". Project A has a dependency on project B, specified as "~>4.5.6". When you're starting developing this project and running "dub" it will download project A with version "1.2.3", then it will download project B with version "4.5.6" as well.

Now, at some later point in time another developer joins your project and clones the project. At the point the developer runs "dub" a new version of project B has been released, "4.5.7". Now dub will download project A with the expected version "1.2.3" and the unexpected version "4.5.7". This can also happen if you're cloning your project on a different computer or a build machine.

Suddenly you have now idea what code you're running and I consider that to be a major security issue. Also I wouldn't trust a single project for a second to not break my code if I updated to a later patch/build version.

We have already gone through this issue in lengthy discussions in the Dub forum.